Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -126,10 +128,34 @@ - (NSString*)debugDescription {
     return [NSString stringWithFormat:@"%@ dataSourceClass: %@", self, [self.dataSource class]];
 }
 
-- (void)refreshVisibleCells {
-    [self.collectionView wmf_enumerateVisibleCellsUsingBlock:
-     ^(WMFArticlePreviewCell* cell, NSIndexPath* path, BOOL* _) {
+#pragma mark - Stay Fresh... yo
+
+- (void)observeArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(articleUpdatedWithNotification:) name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)unobserveArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)articleUpdatedWithNotification:(NSNotification*)note {
+    MWKArticle* article = note.userInfo[MWKArticleKey];
+    [self refreshAnyVisibleCellsWhichAreShowingTitle:article.title];
+}
+
+- (void)refreshAnyVisibleCellsWhichAreShowingTitle:(MWKTitle*)title {
+    NSArray* indexPathsToRefresh = [[self.collectionView indexPathsForVisibleItems] bk_select:^BOOL (NSIndexPath* path) {
+        WMFArticlePreviewCell* cell = (WMFArticlePreviewCell*)[self.collectionView cellForItemAtIndexPath:path];
+        return (cell.title == title);
```

Comment:
this would be ok in Swift, but in ObjC always use `isEqualToTitle:` (or `isEqual:` if a class-specific equality method isn't available). checking **identity** is _usually_ not what you want for model objects. i.e. we want to know if they're equivalent—even if they're different objects.